### PR TITLE
tr2/render: add ps1 lighting contrast modes

### LINKED ISSUE #1646: PS1 contrast levels
This was something Arsunt deliberated on for quite a while but eventually made it happen. He presented it like this:

image

Arsunt:
> DirectX 5 has limited texture lighting formula. Every texture cannot be displayed lighter than it is. 
> But it can be displayed darker than it is. PS1 had full range of texture brightness, so PS1 is more contrast when it comes to dynamic lighting and bright rooms
> So I made 3 modes of lighting contrast:
> - Low: texture lighting is clamped in 0%-100% range (that's how behaved PC version with hardware renderer).
> - Medium: texture lighting is clamped in 0%-150% range (that's most comfort way to play in my opinion, compromise between Low and High).
> - High: texture lighting is clamped in 0%-200% range (that's how behaved PC version with software renderer and PS1 version).
> This is especially noticeable when you use flare or flashing with guns

TR2M before (DX5):
DX5

TR2M after (DX7):
DX7

TR2X:
image

PS1:
image

In some levels that are naturally dark it can be quite striking - compare the dragon:
PC: 
pc-dragon

PS1:
ps1-dragon

### PR DESCRIPTION
**Pull request by rr-**
#### Checklist

- [x] I have read the [coding conventions](https://github.com/LostArtefacts/TR1X/blob/master/CONTRIBUTING.md#coding-conventions)
- [x] I have added a changelog entry about what my pull request accomplishes, or it is an internal change

#### Description

Resolves #1646.
Implementation similar to TR2Main, with 100%/150%/200% brightness caps.

### CODE REVIEW

**Comment by Richard-L** [comment]
😮
I wasn't ready for this, I'm shook!
Can't wait to try the above build.

File: src/tr2/config_map.def
Comment (lahm86) [inline src/tr2/config_map.def]: It may just be my setup but the "low" option is what feels closest to OG PC for me. Should low be the default? Medium and high feel quite dark.

File: src/tr2/game/input.c
Comment (lahm86) [inline src/tr2/game/input.c]: Minor formatting alignment.

File: src/libtrx/include/libtrx/game/input/backends/keyboard_tr2.def
Comment (lahm86) [inline src/libtrx/include/libtrx/game/input/backends/keyboard_tr2.def]: Minor formatting alignment.

File: src/libtrx/include/libtrx/game/input/roles_tr2.def
Comment (lahm86) [inline src/libtrx/include/libtrx/game/input/roles_tr2.def]: Minor formatting alignment.

**Review by lahm86 — COMMENTED** [review]
I'm unable to see a difference between medium and high, as below. If these are the intended shades, could we possibly make Low the default setting? I feel med/high would be too dark to offer as default, and low is closest to OG PC.

20241216_212245_The_Great_Wall
20241216_212248_The_Great_Wall
20241216_212252_The_Great_Wall

On med/high the title screen text is difficult to read.
image

**Comment by rr-** [comment]
@lahm86 did you upgrade the shaders? It's not supposed to look dark.
20241216_203859_Tibetan_Foothills
20241216_203900_Tibetan_Foothills
20241216_203901_Tibetan_Foothills

File: src/libtrx/include/libtrx/game/input/roles_tr2.def
Comment (rr-) [reply src/libtrx/include/libtrx/game/input/roles_tr2.def]: Fixed

File: src/libtrx/include/libtrx/game/input/backends/keyboard_tr2.def
Comment (rr-) [reply src/libtrx/include/libtrx/game/input/backends/keyboard_tr2.def]: Fixed

File: src/tr2/game/input.c
Comment (rr-) [reply src/tr2/game/input.c]: Fixed

**Comment by lahm86** [comment]
My bad, I built locally and didn't copy those over. Can see the difference now, thanks.
I still prefer Low I have to admit 😄 Flares when on the snow hurt my eyes in med/high, but that could just be me.

**Review by lahm86 — APPROVED** [review]

**Comment by rr-** [comment]
> My bad, I built locally and didn't copy those over. Can see the difference now, thanks. I still prefer Low I have to admit 😄 Flares when on the snow hurt my eyes in med/high, but that could just be me.

We need to replace the red eyeglasses with this item:
https://en.wikipedia.org/wiki/Snow_goggles
:D 

I'd leave it as high or maybe medium as middle ground, to increase this feature's visibility. It's more likely that people who dislike it will look for an option to change this, rather than them looking for an option they might not know exists.

**Comment by lahm86** [comment]
> > My bad, I built locally and didn't copy those over. Can see the difference now, thanks. I still prefer Low I have to admit 😄 Flares when on the snow hurt my eyes in med/high, but that could just be me.
> 
> We need to replace the red eyeglasses with this item: https://en.wikipedia.org/wiki/Snow_goggles :D
> 
> I'd leave it as high or maybe medium as middle ground, to increase this feature's visibility. It's more likely that people who dislike it will look for an option to change this, rather than them looking for an option they might not know exists.

Ha, snow goggles would be good :smile:
I think going with medium would be a good move, but happy to go with what the majority think on this. Functionality-wise, it's great 👍

**Comment by rr-** [comment]
Thanks – I've changed it to medium. I'll wait for aredfan's approval before merging.

**Review by aredfan — APPROVED** [review]
Thank you for waiting, I appreciate it. I've tested the different lighting contrast modes and everything LGTM. 👍

**Pull request merged**
